source/Calamari/Kubernetes/Commands/Executors/KustomizeExecutor.cs

```diff
@@ -61,7 +61,12 @@ protected override async Task<IEnumerable<ResourceIdentifier>> ApplyAndGetResour
         void ValidateKubectlVersion(string currentDirectory)
         {
             var commandResult = kubectl.ExecuteCommandAndReturnOutput("version", "--client", "-o", "json");
-            CheckResultForErrors(commandResult, currentDirectory);
+            commandResult.LogErrorsWithSanitizedDirectory(log, currentDirectory);
+            if (commandResult.Result.ExitCode != 0)
```

Comment:
Pulled out of the `CheckResultForErrors` method. The way we are using different exceptions around the place for control flow needs a little more scrutiny.